### PR TITLE
TestPyPI reset blog

### PR DESCRIPTION
**Pull request by ewdurbin**

### CODE REVIEW

File: docs/blog/posts/2024-01-24-test-pypi-groundhogs-day.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+We will introduce a requirement that all project names and usernames
+on TestPyPI will all contain a required prefix of `testpypi-YYYY-`.
+This is intended to disambiguate between the things hosted on TestPyPI
+and "real" PyPI, indicating the temporal nature of the accounts and projects.
+
+For project names this will be transparently applied internal to the codebase
+so that publishing tools will "just work"
+without having to modify configurations like `pyproject.toml` when testing.
+
+Presentation on the website will automatically redirect to the prefixed
```
Comment (miketheman) [inline docs/blog/posts/2024-01-24-test-pypi-groundhogs-day.md:56]: This bit is a little confusing. Do folks have to add prefixes to their projects or not?

**Review by miketheman — APPROVED** [review]
Big fan of this!
Left a few comments and questions, but nothing too critical.

File: docs/blog/posts/2024-01-24-test-pypi-groundhogs-day.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+On February 2, 2024, and each following February 2nd,
+the state of [TestPyPI](https://test.pypi.org) will be reset completely.
+
+This includes **all** user accounts, projects, releases, and release files
+published to TestPyPI.
+
+In order to use TestPyPI all past and new users will need to
+create a new account, enable 2FA, generate new API tokens,
+and update any tools to use their new credentials
+or configure []().
```
[the remark this answers is not in the document]
Comment (ewdurbin) [reply docs/blog/posts/2024-01-24-test-pypi-groundhogs-day.md]: aye. ffe49b7a4

File: docs/blog/posts/2024-01-24-test-pypi-groundhogs-day.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+<!-- more -->
+
+TestPyPI is _not_ intended to be a reliable long-term place for publishing
+and particularly not for installing Python packages. It is intended to be a
+sandbox for the community to test their publication tooling and preview the
+result.
+
+Additionally TestPyPI itself has grown significantly to the point that it
+needlessly consumes substantial resources.
```
[the remark this answers is not in the document]
Comment (ewdurbin) [reply docs/blog/posts/2024-01-24-test-pypi-groundhogs-day.md:44]: I don't think we need to dissuade it, but rather just make clear it shouldn't be in critical path.

File: docs/blog/posts/2024-01-24-test-pypi-groundhogs-day.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+## What will change?
+
+We will introduce a requirement that all project names and usernames
+on TestPyPI will all contain a required prefix of `testpypi-YYYY-`.
```
[the remark this answers is not in the document]
Comment (ewdurbin) [reply docs/blog/posts/2024-01-24-test-pypi-groundhogs-day.md:49]: I'd be emptying the object store for test pypi.

File: docs/blog/posts/2024-01-24-test-pypi-groundhogs-day.md (hunk 1 above)
Comment (ewdurbin) [reply docs/blog/posts/2024-01-24-test-pypi-groundhogs-day.md:57]: people would only need to add the prefix to install from testpypi

File: docs/blog/posts/2024-01-24-test-pypi-groundhogs-day.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+so that publishing tools will "just work"
+without having to modify configurations like `pyproject.toml` when testing.
+
+Presentation on the website will automatically redirect to the prefixed
+name, but in order to _install_ from test.pypi.org users _must_ use the prefix.
+
+This will hopefully reduce confusion between the two services and mitigate
+impersonation and unintended usage of TestPyPI due to misconfiguration.
+
+## Why Grounhog Day?
```
Comment (sethmlarson) [inline docs/blog/posts/2024-01-24-test-pypi-groundhogs-day.md:63]: typo

File: docs/blog/posts/2024-01-24-test-pypi-groundhogs-day.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+## What is happening?
+
+On February 2, 2024, and each following February 2nd,
+the state of [TestPyPI](https://test.pypi.org) will be reset completely.
+
+This includes **all** user accounts, projects, releases, and release files
+published to TestPyPI.
+
+In order to use TestPyPI all past and new users will need to
```
Comment (sethmlarson) [inline docs/blog/posts/2024-01-24-test-pypi-groundhogs-day.md:20]: Deleting all projects seems like a good idea, but deleting all users would make any automated usage of TestPyPI frustrating? You'd need to create a new account and tokens/TPs every year per project.

Many project CI's use TestPyPI to check if their package can be uploaded (due to package metadata requirements), if we're intending to remove that use-case we should recommend an alternative for folks in this post (Is `twine --check` sufficient these days?)

**Pull request closed without merging**